File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n = len(temperatures)
        res = [0] * n
        stack = []

        for i, v in enumerate(temperatures):
            while stack and v > temperatures[stack[-1]]:
                temp = stack.pop()
                res[temp] = i - temp  
            stack.append(i)
        
        return res

File: 0739-daily-temperatures/0739-daily-temperatures.go
func dailyTemperatures(temperatures []int) []int {
    res := make([]int, len(temperatures))
    stack := []int{}
    for i, v := range temperatures {
        for len(stack) > 0 && v > temperatures[stack[len(stack)-1]] {
            res[stack[len(stack)-1]] = i - stack[len(stack)-1]
            stack = stack[:len(stack)-1]
        }
        
        stack = append(stack, i)
    }

    return res
}